flink-ai-flow/ai_flow/api/context_extractor.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from abc import ABC, abstractmethod
+from typing import Text
+
+from notification_service.base_notification import BaseEvent
+
+
+class ContextExtractor(ABC):
+    """
+    ContextExtractor is used to decide if a event should be broadcast or extract context from a event.
```

Comment:
ContextExtractor is written by users to decide if an event should be broadcast or we should extract context from the event.

Reply:
fixed in c1785cf